views/src/Pages/ForgotPassword.jsx

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useState, useRef } from "react";
+import {
+  Stepper,
+  Button,
+  Group,
+  TextInput,
+  Container,
+  PinInput,
+} from "@mantine/core";
+import axios from "axios";
+
+import { baseUrl } from "../API/index";
+
+
+
+function ForgotPassword() {
+  const [active, setActive] = useState(0);
+  const [email, setEmail] = useState("");
+  const [emailSubmitted, setEmailSubmitted] = useState(false);
+  const [otpValidated, setOtpValidated] = useState(false);
+  const errorMsg = useRef(null);
+  const [otp, setOtp] = useState(0);
+  const [newPwd, setNewPwd] = useState(null);
+  const [confirmPwd, setConfirmPwd] = useState(null);
+  const updatepasswordValidator = (input, confirmation) => {
+    errorMsg.current.textContent = ``;
+    let newPwd = input;
+    let confirmPwd = confirmation;
+    if (!newPwd || !confirmPwd) {
+      errorMsg.current.textContent =
+        "Please enter and confirm your new password";
+    }
+    if (typeof newPwd !== "string") {
+      errorMsg.current.textContent = "New password must be a string";
+    }
+    if (newPwd !== confirmPwd) {
+      errorMsg.current.textContent = "Passwords do not match";
+    }
+    if (newPwd.length < 8) {
+      errorMsg.current.textContent =
+        "Password must be at least 8 characters long";
+    }
+
+    if (newPwd.length > 30) {
+      errorMsg.current.textContent =
+        "Password cannot be more than 30 characters long";
+    }
+  };
+  const nextStep = async () => {
```

Comment:
This whole function has too many nested if else statement making it difficult to understand.  
I suggest either breaking out the various steps into their own functions and using them or refactoring so we
don't have too many if else nested. Look into guard clauses and use them where they apply.

Reply:
Implemented simpler syntax so the max depth is 2 nested statements